clients/src/test/java/org/apache/kafka/clients/MetadataTest.java

```diff
@@ -105,6 +105,19 @@ private static void checkTimeToNextUpdate(long refreshBackoffMs, long metadataEx
         assertEquals(0, metadata.timeToNextUpdate(now + 1));
     }
 
+    @Test
+    public void testUpdateMetadataAllowedImmediatelyAfterBootstrap() {
+        MockTime time = new MockTime();
+
+        Metadata metadata = new Metadata(refreshBackoffMs, metadataExpireMs, new LogContext(),
+                new ClusterResourceListeners());
+        metadata.bootstrap(Collections.singletonList(new InetSocketAddress("localhost", 9002)));
+
+        metadata.requestUpdate();
```

Comment:
`bootstrap()` itself should mandate an update, right? Do we need to call `requestUpdate`?